config/opal_mca.m4

```diff
@@ -167,29 +167,34 @@ of type-component pairs.  For example, --enable-mca-no-build=pml-ob1])
     # Second, set the DSO_all and STATIC_all variables.  conflict
     # resolution (prefer static) is done in the big loop below
     #
+    # Exception here is the components of the accelerator framework,
+    # which by default are built to be dynamic.
+    #
     AC_MSG_CHECKING([which components should be run-time loadable])
     if test "$enable_static" != "no"; then
         DSO_all=0
         msg="none (static libraries built)"
     elif test "$OPAL_ENABLE_DLOPEN_SUPPORT" = 0; then
         DSO_all=0
         msg="none (dlopen disabled)"
-    elif test -z "$enable_mca_dso"; then
-        DSO_all=0
-        msg=default
     elif test "$enable_mca_dso" = "no"; then
         DSO_all=0
         msg=none
     elif test "$enable_mca_dso" = "yes"; then
         DSO_all=1
         msg=all
     else
+        if test -z "$enable_mca_dso"; then
+            enable_mca_dso="accelerator,btl-smcuda,rcache-gpusm,rcache-rgpusm"
+            AC_MSG_NOTICE([building accelerator framework compnents as DSOs (default)])
```

Comment:
Putting this `AC_MSG_NOTICE` here will interrupt the `AC_MSG_CHECKING` / `AC_MSG_RESULT`.

Maybe you should just assign `msg` here with some kind of suffix to put after the list of components in the `AC_MSG_RESULT`...?  Perhaps something like (just typed here in github; not tested):

```sh
        msg=
        if test -z "$enable_mca_dso"; then
            # We set all accelerator-related components to build as DSO by default.
            # See ...URL of github issue or something with more explanation...
            enable_mca_dso="accelerator,btl-smcuda,rcache-gpusm,rcache-rgpusm"
            msg="(default)"
        fi
        DSO_all=0
        ifs_save="$IFS"
        IFS="${IFS}$PATH_SEPARATOR,"
        for item in $enable_mca_dso; do
            AS_VAR_SET([AS_TR_SH([DSO_$item])], [1])
            str="`echo DSO_$item=1 | sed s/-/_/g`"
            eval $str
            msg="$item $msg"
        done
        IFS="$ifs_save"
    fi
    AC_MSG_RESULT([$msg])

```

Reply:
that sounds like a good idea.  fixing.